File: luka_alhonen_ex05/p02/medians.c
#include <stdio.h>

int main(int argc, char** argv)
{
	FILE* file;
	double value, sum;
	int count;

	for (int i = 1; i < argc; i++) {
		char* filename = argv[i];
		file = fopen(filename, "r");
		if (file != NULL) {
			value = 0.0;
			sum = 0.0;
			count = 0;

			while(fscanf(file, "%lf", &value) == 1) {
				sum += value;
				count++;
			}

			if (count > 0) {
				double average = sum / count;
				printf("Average of %s: %f\n", filename, average);
			}
		}
		fclose(file);
	}

	return 0;
}

File: luka_alhonen_ex05/p02/medians.py
import numpy as np
import sys

for file in sys.argv[1:]:
    total_time = 0.0
    count = 0
    with open(file, "r") as a_out:
        for line in a_out:
            words = line.split()
            cpu_time = float(words[1])
            total_time += cpu_time
            count += 1
        
        print(f'Average of {file.replace("_output.txt", "")}: {total_time / count} s')